File: trunk/core/src/main/java/org/runetranscriber/core/viking/example/AngloSaxonExample2.java
package org.runetranscriber.core.viking.example;

import org.runetranscriber.core.AbstractExample;
import org.runetranscriber.core.FontLetterList;
import org.runetranscriber.core.LanguageLetterList;
import org.runetranscriber.core.Phoneme;
import org.runetranscriber.core.PhonemeList;
import org.runetranscriber.core.RuneList;
import org.runetranscriber.core.viking.AngloSaxonFontLetter;
import org.runetranscriber.core.viking.AngloSaxonRune;

/**
 * Provides an example for Anglo-Saxon runes from The Hobbit, Thror's Map #1.
 */
public final class AngloSaxonExample2 extends AbstractExample<AngloSaxonRune, AngloSaxonFontLetter>
{
    @Override
    protected FontLetterList<AngloSaxonFontLetter> createFontLetters()
    {
        final AngloSaxonFontLetter[] myFontLetters = { AngloSaxonFontLetter.EH, AngloSaxonFontLetter.SIGEL,
                AngloSaxonFontLetter.WYNN, AngloSaxonFontLetter.NYD, };

        return new FontLetterList<AngloSaxonFontLetter>(myFontLetters);
    }

    @Override
    protected LanguageLetterList createLanguageLetters()
    {
        final String text = "ESWN";

        return new LanguageLetterList(text);
    }

    /**
     * @return phonemes.
     */
    @Override
    protected PhonemeList createPhonemes()
    {
        final Phoneme[] myPhonemes = {
                // e
                Phoneme.E,
                // s
                Phoneme.S,
                // w
                Phoneme.W,
                // n
                Phoneme.N, };

        return new PhonemeList(myPhonemes);
    }

    /**
     * @return runes.
     */
    @Override
    protected RuneList<AngloSaxonRune> createRunes()
    {
        final AngloSaxonRune[] myRunes = {
                // e
                AngloSaxonRune.EH,
                // s
                AngloSaxonRune.SIGEL,
                // w
                AngloSaxonRune.WYNN,
                // n
                AngloSaxonRune.NYD, };

        return new RuneList<AngloSaxonRune>(myRunes);
    }
}

File: webapp/src/main/js/example/AngloSaxonExample2.js
/*
 * Provides an example for Anglo-Saxon runes from The Hobbit, Thror's Map #1.
 */
define(
        [ "AngloSaxonRune" ],
        function(AngloSaxonRune)
        {
            "use strict";
            var AngloSaxonExample2 =
            {
                NAME: "Anglo-Saxon Example 2",

                IMAGE: "https://rawgit.com/jmthompson2015/runetranscriber/master/core/doc/images/hobbit/TheHobbit_ThrorsMap1EastTop.jpg",
                IMAGE_WIDTH: 200,

                LANGUAGE_STRING: "E S W N",

                LANGUAGE_WORDS: [ "e", " ", "s", " ", "w", " ", "n" ],

                PHONEMES: [ AngloSaxonRune.properties["eh"].phoneme, AngloSaxonRune.properties["space"].phoneme,
                        AngloSaxonRune.properties["sigel"].phoneme, AngloSaxonRune.properties["space"].phoneme,
                        AngloSaxonRune.properties["wynn"].phoneme, AngloSaxonRune.properties["space"].phoneme,
                        AngloSaxonRune.properties["nyd"].phoneme ],

                RUNES: [
                // e
                AngloSaxonRune.EH, AngloSaxonRune.SPACE,
                // s
                AngloSaxonRune.SIGEL, AngloSaxonRune.SPACE,
                // w
                AngloSaxonRune.WYNN, AngloSaxonRune.SPACE,
                // n
                AngloSaxonRune.NYD, ],

                FONT_LETTERS: [ AngloSaxonRune.properties["eh"].fontLetter,
                        AngloSaxonRune.properties["space"].fontLetter, AngloSaxonRune.properties["sigel"].fontLetter,
                        AngloSaxonRune.properties["space"].fontLetter, AngloSaxonRune.properties["wynn"].fontLetter,
                        AngloSaxonRune.properties["space"].fontLetter, AngloSaxonRune.properties["nyd"].fontLetter ],
            };

            if (Object.freeze)
            {
                Object.freeze(AngloSaxonExample2);
            }

            return AngloSaxonExample2;
        });
